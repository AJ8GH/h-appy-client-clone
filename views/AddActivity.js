/* eslint-disable react/prop-types */
import React, { useState } from "react";
import Slider from "@react-native-community/slider";
import {
  Text, View, TextInput, StyleSheet, TouchableOpacity, Picker,
} from 'react-native';
// import { Picker } from '@react-native-picker/picker';
import { useNavigation } from '@react-navigation/native';
import Header from '../components/Header';
import { storeData, doesActivityNameExist, addToUserData } from '../src/UserData';
import { FetchCategories } from "../src/FetchActivities";

export default function AddActivity() {
  const [ActivityType, setActivityType] = useState("default");
  const [ActivityName, setActivityName] = useState("");
  const [accessibility, setAccessibility] = useState(0);
  const [price, setPrice] = useState(0);
  const [category, setCategory] = useState([]);

  const categories = FetchCategories();

  return (
    <View style={styles.container}>
      <Header />

      <View style={styles.FormContainer}>

        <View style={styles.nameContainer}>
          <TextInput
            editable
            style={styles.nameField}
            placeholder="Activity name..."
            placeholderTextColor="#fff"
            onChangeText={(newActivityName) => setActivityName(newActivityName)}
            defaultValue={ActivityName}
          />
        </View>

        <View style={styles.pickerContainer}>
          <Picker
            style={styles.picker}
            selectedValue={ActivityType}
            onValueChange={(itemValue, itemIndex) => setActivityType(itemValue)}
          >
            <Picker.Item
              label="Select a Menu Section..."
              value="default"
              enabled={false}
            />
            <Picker.Item label="Nibbles" value="nibbles" />
            <Picker.Item label="Appetisers" value="appetisers" />
            <Picker.Item label="Mains" value="mains" />
            <Picker.Item label="Desserts" value="desserts" />
          </Picker>
        </View>

        <View style={styles.pickerContainer}>
          <Picker
            style={styles.picker}
            selectedValue={category}
            onValueChange={(itemValue, itemIndex) => setCategory(itemValue)}
          >
            <Picker.Item
              label="Select a Category..."
              value="default"
              enabled={false}
            />
            {categories.map((item, index) => {
              return <Picker.Item label={item} value={item} key={index} />;
            })}
          </Picker>
        </View>

        <AccessibilitySlider
          accessibility={accessibility}
          setAccessibility={setAccessibility}
        />

        <PriceSlider price={price} setPrice={setPrice} />

      </View>
      <SubmitButton
        ActivityType={ActivityType}
        ActivityName={ActivityName}
        accessibility={accessibility}
        price={price}
        categories={[category]}
      />
    </View>
  );
}

function AccessibilitySlider(props) {
  const { accessibility, setAccessibility } = props;
  return (
    <View style={styles.sliderContainer}>
      <Slider
        style={styles.slider}
        minimumValue={0}
        maximumValue={10}
        minimumTrackTintColor="#696773"
        maximumTrackTintColor="#363946"
        onValueChange={(value) => setAccessibility(Math.ceil(value))}
      />
      <Text style={styles.sliderText}>
        Accessibility Score: {'\t'} {accessibility}
      </Text>
    </View>
  );
}

function PriceSlider(props) {
  const { price, setPrice } = props;
  return (
    <View style={styles.sliderContainer}>
      <Slider
        style={styles.slider}
        minimumValue={0}
        maximumValue={4}
        minimumTrackTintColor="#696773"
        maximumTrackTintColor="#363946"
        onValueChange={(value) => setPrice(Math.ceil(value))}
        />
        <Text style={styles.sliderText}>
          Cost: {'\t'} {'Â£ '.repeat(price) || 'Free :)'}
        </Text>
    </View>
  );
}

function SubmitButton(props) {
  const navigation = useNavigation();
  const {
    ActivityType,
    ActivityName,
    accessibility,
    price,
    categories,
  } = props;
  return (
    <View style={styles.submitButtonContainer}>
      <TouchableOpacity
        onPress={() => {
          if (ActivityType === 'default') {
            alert('Please select an activity type');
          } else if (ActivityName === '') {
            alert('Please enter an activity name!');
          } else if (doesActivityNameExist(ActivityType, ActivityName)) {
            alert('Activity name already exists!');
          } else {
            storeData(ActivityType, {
              _id: ActivityName, name: ActivityName, accessibility, price,
            });
            navigation.navigate('Menu');
          }
        }}
      >
        <Text style={styles.submitButton}>Add</Text>
      </TouchableOpacity>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#696773",
    // alignContent: 'flex-start',
  },
  FormContainer: {
    flex: 0.82,
    width: "92%",
    marginTop: 10,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: "#B1B6A6",
    borderRadius: 5,
  },
  nameContainer: {
    flex: 0.15,
    borderRadius: 5,
    marginTop: "5%",
    padding: 10,
    width: "90%",
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: '#363946',
  },
  nameField: {
    color: "#fff",
    fontFamily: 'Chalkduster',
    fontSize: 18,
  },
  slider: {
    width: 280,
    opacity: 1,
    height: 50,
  },
  sliderContainer: {
    flex: 0.16,
    alignItems: "center",
    justifyContent: "center",
    backgroundColor: '#819595',
    width: '90%',
    borderRadius: 5,
    marginBottom: 15,
  },
  sliderText: {
    color: '#23252E',
    fontFamily: 'Courier',
    fontSize: 16
  },
  pickerContainer: {
    marginTop: (Platform.OS === 'ios') ? -50 : 0,
    width: "80%",
    borderRadius: 10,
    flex: 0.3,
  },
  picker: {
    transform: (Platform.OS === 'ios') ? [{ scaleX: 0.80 }, { scaleY: 0.80 }] : [],
  },
  submitButtonContainer: {
    width: "92%",
    marginLeft: 8,
    paddingTop: 10,
    paddingBottom: 10,
    backgroundColor: "#363946",
    justifyContent: "center",
    alignItems: "center",
    borderRadius: 8,
    position: "absolute",
    bottom: 15,
  },
  submitButton: {
    color: "#B1B6A6",
    fontFamily: 'Courier',
    fontSize: 20,
  },
});
