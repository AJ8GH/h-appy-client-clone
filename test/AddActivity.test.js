import React from 'react';
import renderer from 'react-test-renderer';
import AddActivity from '../views/AddActivity';

test('renders correctly', () => {
  const tree = renderer.create(<AddActivity />).toJSON();
  expect(tree).toMatchSnapshot();
});
