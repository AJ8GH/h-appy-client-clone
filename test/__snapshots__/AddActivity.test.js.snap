// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignContent": "flex-start",
      "alignItems": "center",
      "backgroundColor": "#f8f9d4",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#c7524a",
        "flex": 1,
        "fontSize": 24,
        "fontWeight": "bold",
        "height": 60,
        "justifyContent": "center",
        "maxHeight": 60,
        "position": "absolute",
        "top": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "alignItems": "center",
          "fontFamily": "Didot",
          "fontSize": 25,
          "fontWeight": "bold",
          "justifyContent": "center",
        }
      }
    >
      H-Appy
    </Text>
  </View>
  <Text>
     This will be we can add activities as a user 
  </Text>
  <View
    style={
      Object {
        "borderWidth": 2,
        "margin": 3,
        "width": "80%",
      }
    }
  >
    <Text>
      Activity Type
    </Text>
  </View>
  <View
    style={
      Object {
        "borderWidth": 2,
        "margin": 3,
        "width": "80%",
      }
    }
  >
    <Text>
      Activity Name
    </Text>
    <TextInput
      allowFontScaling={true}
      defaultValue=""
      onChangeText={[Function]}
      placeholder="enter the name of the activity here"
      rejectResponderTermination={true}
      style={
        Object {
          "height": 40,
        }
      }
      underlineColorAndroid="transparent"
    />
  </View>
  <View>
    <Text>
      Accessibility:
      0
    </Text>
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumTrackTintColor="#000000"
      maximumValue={10}
      minimumTrackTintColor="#FFFFFF"
      minimumValue={0}
      onChange={null}
      onRNCSliderSlidingComplete={[Function]}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={null}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={0}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "height": 50,
            "marginTop": 50,
            "opacity": 1,
            "width": 300,
          },
        ]
      }
      thumbImage={
        Object {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        }
      }
      value={0}
    />
  </View>
  <View>
    <Text>
      Price:
      0
    </Text>
    <RNCSlider
      disabled={false}
      enabled={true}
      inverted={false}
      maximumTrackTintColor="#000000"
      maximumValue={10}
      minimumTrackTintColor="#FFFFFF"
      minimumValue={0}
      onChange={null}
      onRNCSliderSlidingComplete={[Function]}
      onRNCSliderSlidingStart={null}
      onRNCSliderValueChange={null}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      step={0}
      style={
        Array [
          Object {
            "height": 40,
          },
          Object {
            "height": 50,
            "marginTop": 50,
            "opacity": 1,
            "width": 300,
          },
        ]
      }
      thumbImage={
        Object {
          "__packager_asset": true,
          "height": 1,
          "scale": 1,
          "uri": "file://assets/full/path/to/directory/name.type",
          "width": 1,
        }
      }
      value={0}
    />
  </View>
</View>
`;
